File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/06.Nested-Loop/02-exercise/06.cinemaTickets.js
function cinemaTickets(input) {

    let index = 0;
    let current = input[index++];

    let totalTickets = 0;
    let studentTicketCounter = 0;
    let standardTicketCounter = 0;
    let kidTicketCounter = 0;

    while (current !== "Finish") {
        let places = Number(input[index++]);
        let currentTicket = input[index++];
        let currentFilmTicketsCounter = 0;
        while (currentTicket !== "End") {
            currentFilmTicketsCounter++;
            totalTickets++;
            switch (currentTicket) {
                case "student": studentTicketCounter++; break;
                case "standard": standardTicketCounter++; break;
                case "kid": kidTicketCounter++; break;
            }
            if (places - currentFilmTicketsCounter  === 0) {
                break;
            }
            
            currentTicket = input[index++];
        }
        const percentageForCurrentFilm = currentFilmTicketsCounter / places * 100;
        console.log(`${current} - ${percentageForCurrentFilm.toFixed(2)}% full.`);

        current = input[index++];
    }

    const studentTicketsPrecentage = studentTicketCounter / totalTickets * 100;
    const standardTicketsPrecentage = standardTicketCounter / totalTickets * 100;
    const kidTicketsPrecentage = kidTicketCounter / totalTickets * 100;
    console.log(`Total tickets: ${totalTickets}`);
    console.log(`${studentTicketsPrecentage.toFixed(2)}% student tickets.`);
    console.log(`${standardTicketsPrecentage.toFixed(2)}% standard tickets.`);
    console.log(`${kidTicketsPrecentage.toFixed(2)}% kids tickets.`);
}
cinemaTickets(["The Matrix",
"20",
"student",
"standard",
"kid",
"kid",
"student",
"student",
"standard",
"student",
"End",
"The Green Mile",
"17",
"student",
"standard",
"standard",
"student",
"standard",
"student",
"End",
"Amadeus",
"3",
"standard",
"standard",
"standard",
"Finish"]);

File: Python_Basics/Homeworks-And-Labs/06.nested_loops/02.exercise/06.cinema_tickets.py
current = input()

total_tickets = 0
student_ticket_counter = 0
standard_ticket_counter = 0
kid_ticket_counter = 0

while current != "Finish":
    places = int(input())
    current_ticket = input()
    current_film_tickets_counter = 0
    while current_ticket != "End":
        current_film_tickets_counter += 1
        total_tickets += 1
        if current_ticket == 'student':
            student_ticket_counter += 1
        elif current_ticket == 'standard':
            standard_ticket_counter += 1
        elif current_ticket == 'kid':
            kid_ticket_counter += 1
            
        if places - current_film_tickets_counter  == 0:
            break 
    
        current_ticket = input()
        
    percentage_for_current_film = current_film_tickets_counter / places * 100
    print(f'{current} - {percentage_for_current_film:.2f}% full.')

    current = input()

student_tickets_percentage = student_ticket_counter / total_tickets * 100
standard_tickets_percentage = standard_ticket_counter / total_tickets * 100
kid_tickets_percentage = kid_ticket_counter / total_tickets * 100
print(f'Total tickets: {total_tickets}')
print(f'{student_tickets_percentage:.2f}% student tickets.')
print(f'{standard_tickets_percentage:.2f}% standard tickets.')
print(f'{kid_tickets_percentage:.2f}% kids tickets.')
